### ISSUE TITLE
Thumbnails not generating correctly

### ISSUE THREAD

**Issue by ecrocombe** [nopSolutions/nopCommerce#5948]
nopCommerce version: 4.40.4

Steps to reproduce the problem: 
1. Switch to store media in "files" instead of "database" (I'm not even sure if this is required)
2. Create product with image: https://images.icecat.biz/img/gallery_raw/92870481_9502944695.png
3. Observe product in Admin Product List ![sampleofthumb](https://user-images.githubusercontent.com/42088209/141652371-4fca4e6f-9616-4574-9a0c-748883fd096b.png)
4. Observe product in Public Product Page ![sampleofthumbinpublic](https://user-images.githubusercontent.com/42088209/141652508-24168325-1c0d-4527-a5d4-4c6a40cc0cb0.png)
5. Although in the sample pictures above you can see I'm not using the default theme, the issue also is reproducible using the default Clean theme.
6. I believe this issue is also causing SQL Command Timeouts (although I cannot confirm at this time if its related) when viewing products in category on the public site when multiple products have the same type of issue with pictures (Category Page size:30). 

```
System.Data.SqlClient.SqlException` (0x80131904): Timeout expired.  The timeout period elapsed prior to completion of the operation or the server is not responding.
---> System.ComponentModel.Win32Exception (258): The wait operation timed out.
   at System.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__126_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location ---
   at LinqToDB.Data.DataConnection.ExecuteReaderAsync(CommandBehavior commandBehavior, CancellationToken cancellationToken)
   at LinqToDB.Data.DataConnection.QueryRunner.ExecuteReaderAsync(CancellationToken cancellationToken)
   at LinqToDB.Linq.QueryRunner.ExecuteQueryAsync[T](Query query, IDataContext dataContext, Mapper`1 mapper, Expression expression, Object[] ps, Object[] preambles, Int32 queryNumber, Func`2 func, TakeSkipDelegate skipAction, TakeSkipDelegate takeAction, CancellationToken cancellationToken)
   at LinqToDB.Linq.ExpressionQuery`1.GetForEachAsync(Action`1 action, CancellationToken cancellationToken)
   at LinqToDB.AsyncExtensions.ToListAsync[TSource](IQueryable`1 source, CancellationToken token)
   at System.Linq.AsyncIQueryableExtensions.ToPagedListAsync[T](IQueryable`1 source, Int32 pageIndex, Int32 pageSize, Boolean getOnlyTotalCount)
   at Nop.Services.Catalog.ProductService.SearchProductsAsync(Int32 pageIndex, Int32 pageSize, IList`1 categoryIds, IList`1 manufacturerIds, Int32 storeId, Int32 vendorId, Int32 warehouseId, Nullable`1 productType, Boolean visibleIndividuallyOnly, Boolean excludeFeaturedProducts, Nullable`1 priceMin, Nullable`1 priceMax, Int32 productTagId, String keywords, Boolean searchDescriptions, Boolean searchManufacturerPartNumber, Boolean searchSku, Boolean searchProductTags, Int32 languageId, IList`1 filteredSpecOptions, ProductSortingEnum orderBy, Boolean showHidden, Nullable`1 overridePublished)
   at Nop.Web.Factories.CatalogModelFactory.<>c__DisplayClass45_0.<<PrepareCategoryProductsModelAsync>g__getProductPriceAsync|1>d.MoveNext()
--- End of stack trace from previous location ---
   at Nop.Web.Factories.CatalogModelFactory.PrepareCategoryProductsModelAsync(Category category, CatalogProductsCommand command)
   at Nop.Web.Factories.CatalogModelFactory.PrepareCategoryModelAsync(Category category, CatalogProductsCommand command)
   at Nop.Web.Controllers.CatalogController.Category(Int32 categoryId, CatalogProductsCommand command)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Logged|17_1(ResourceInvoker invoker)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at WebMarkupMin.AspNetCore5.WebMarkupMinMiddleware.ProcessAsync(HttpContext context, Boolean useMinification, Boolean useCompression)
   at WebMarkupMin.AspNetCore5.WebMarkupMinMiddleware.ProcessAsync(HttpContext context, Boolean useMinification, Boolean useCompression)
   at WebMarkupMin.AspNetCore5.WebMarkupMinMiddlewareBase.Invoke(HttpContext context)
   at StackExchange.Profiling.MiniProfilerMiddleware.Invoke(HttpContext context) in C:\projects\dotnet\src\MiniProfiler.AspNetCore\MiniProfilerMiddleware.cs:line 125
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Nop.Services.Authentication.AuthenticationMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Session.SessionMiddleware.Invoke(HttpContext context)
   at Nop.Services.Installation.InstallUrlMiddleware.InvokeAsync(HttpContext context, IWebHelper webHelper)
   at Nop.Services.Common.KeepAliveMiddleware.InvokeAsync(HttpContext context, IWebHelper webHelper)
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
ClientConnectionId:14c55bbc-8e71-4749-a105-87e1fb8e76ec
Error Number:-2,State:0,Class:11
```

~~My store is still under development, therefore I cannot guarantee access to the below URL's, but here they are anyway:~~

~~Public Category containing multiple products with the issue and is causing timeouts:~~
~~- https://shop.veretech.systems/notebooks~~

~~Public Product Pages where problematic images seem to exist (this is only a few):~~
~~- https://shop.veretech.systems/d515ua-r7-5700u-8-512-fhd-vips-win10~~
~~- https://shop.veretech.systems/d515ua-r5-5500u-8-512-fhd-vips-win10~~
~~- https://shop.veretech.systems/gv301qc-r9-5900hs-rtx3050-8gx2-512-win10~~
~~- https://shop.veretech.systems/x515ea-i5-1135g7-8-512-fhd-vips-win10~~
~~- https://shop.veretech.systems/x515ea-i5-1135g7-8-256-fhd-tn-win10~~
~~- https://shop.veretech.systems/x515epi7-1165g7-mx330-8-512-fhd-vips-w10~~
~~- https://shop.veretech.systems/x515ep-i5-1135g7-mx330-8-512-fhdvips-w10~~

EDIT:

1. Formatted exception into its own block
2. [Strike through URL's which are no longer valid](https://github.com/nopSolutions/nopCommerce/issues/5948#issuecomment-968621941)

**Comment by ecrocombe** [comment]
Unfortunately I have had to change product images and the URL's above are no longer showing the issue. The issue with the thumbnails generating large images (or returning the originally uploaded image) is still a concern though, as the images on my site are imported from an external source automatically. I have patched this by redrawing the original image to something a wee bit smaller in code before uploading.

As for the SQL Command Timeouts, I believe it was due to ~900+ products in a single category with many different filterable Specification Attributes. I have temporarily fixed this by further separating products into different categories, reducing both the count of products and count of Filterable Specification Attributes in the category causing the issue. I'm almost positive this is an issue on its own, but that's for another day.
